spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/util/SerializationUtils.java

```diff
@@ -8,6 +8,10 @@
 
 import org.springframework.core.ConfigurableObjectInputStream;
 
+@Deprecated
```

Comment:
I'm debating whether we should deprecate. I feel there are a lot of touch points with `SerializationStrategy` via `setSerializationStrategy()` in the various collaborators. My preference would be to touch less of the code. Since all collaborators use `SerializationUtils` it might be better to isolate the changes there.

We could add a `SerializationStrategy` static member and default to `WhitelistedSerializationStrategy`. The application could override the behaviour via these options:

1. Add new `public static void setSerializationStrategy(SerializationStrategy serializationStrategy)`
2. Using `spring.factories`

With `spring.factories`, the user could override by adding a classpath resource under `META-INF/spring.factories` with the contents being:

`org.springframework.security.oauth2.common.util.SerializationStrategy = mypackage.CustomSerializationStrategy`

You can use `SpringFactoriesLoader` to check if the user has supplied a custom `SerializationStrategy`.

What are your thoughts on this?

Reply:
Okay, let me try to add a `SerializationStrategy` static member to `SerializationUtils` and a new  `SerializationUtils.setSerializationStrategy()` method. Looks like it can make simplify the patch which is definitely good.

I am planning to get back to it on Nov 4th. I think we have time to finish it the next week. Thanks for the comments!